app/components/citydata-sample/citydata.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component } from '@angular/core';
+import { Http, Response, URLSearchParams } from '@angular/http';
+import { CONSTS } from '../../config/constants';
+import { Observable }     from 'rxjs/Observable';
+
+@Component({
+    selector: 'city-data',
+    templateUrl: './citydata.component.html',
+    styleUrls: ['./citydata.component.less']
+})
+
+export class CityDataInfoComponent {
+    private weatherURL = CONSTS.WEATHER_URL;
+    private API_KEY = CONSTS.API_KEY;
+    private endPoint: string = 'weather';
+    cityName: string;
+    cityInfo: Observable<number>;
+    constructor(private http: Http) {}
+
+    getCityWeather(city: string) {
+        console.log(city);
+        this.cityName = city;
+        this.cityInfo = this.getObservableCityWeather(city);
```

Comment:
It's better to avoid mentioning types in functions names. Using types is really enough.